ui/src/app/store/machine/utils.ts

```diff
@@ -69,6 +70,33 @@ export const useCanEdit = (
   );
 };
 
+/**
+ * Check whether a machine's status allows storage configuration.
+ * @param machine - A machine object.
+ * @returns Whether the machine's status allows storage configuration.
+ */
+export const useMachineStorageConfigurable = (
```

Comment:
I don't think these need to be hooks as they don't do anything that a normal function can't.

Reply:
Ah you're right. Originally I had them extending `useCanEdit` but it turns out I needed them to be separate.